File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/parking/GetParkingFeeForFleetUseCase.java
package com.lattis.ellipse.domain.interactor.parking;

import com.lattis.ellipse.data.network.model.response.parking.GetParkingFeeForFleetResponse;
import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.model.Location;
import com.lattis.ellipse.domain.repository.ParkingRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 8/1/17.
 */

public class GetParkingFeeForFleetUseCase extends UseCase<GetParkingFeeForFleetResponse> {

    private ParkingRepository parkingRepository;
    private Location location;
    private int fleetId;

    @Inject
    public GetParkingFeeForFleetUseCase(ThreadExecutor threadExecutor,
                                      PostExecutionThread postExecutionThread,
                                      ParkingRepository parkingRepository) {
        super(threadExecutor, postExecutionThread);
        this.parkingRepository = parkingRepository;
    }

    public GetParkingFeeForFleetUseCase withFleetId(int fleetId) {
        this.fleetId = fleetId;
        return this;
    }

    public GetParkingFeeForFleetUseCase withLocation(Location location) {
        this.location = location;
        return this;
    }

    @Override
    protected Observable<GetParkingFeeForFleetResponse> buildUseCaseObservable() {
        return parkingRepository.getParkingFeeForFleet(location,fleetId);
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/parking/GetParkingFeeForFleetUseCase.kt
package com.lattis.domain.usecase.parking

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.models.ParkingFeeForFleet
import com.lattis.domain.repository.ParkingRepository
import com.lattis.domain.usecase.base.UseCase
import com.lattis.domain.models.Location
import com.lattis.domain.models.Ride
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class GetParkingFeeForFleetUseCase @Inject
constructor(threadExecutor: ThreadExecutor,
            postExecutionThread: PostExecutionThread,
            private val parkingRepository: ParkingRepository
) : UseCase<ParkingFeeForFleet>(threadExecutor, postExecutionThread) {

    private var location: Location? = null
    private var fleetId = 0

    fun withFleetId(fleetId: Int): GetParkingFeeForFleetUseCase {
        this.fleetId = fleetId
        return this
    }

    fun withLocation(location: Location): GetParkingFeeForFleetUseCase {
        this.location = location
        return this
    }

    override fun buildUseCaseObservable(): Observable<ParkingFeeForFleet> {
        return this.parkingRepository.getParkingFeeForFleet(location!!,fleetId)
    }
}